File: a1/q2/BankCli.java
//Created by:	Colin Dyson
//Student #:	7683407

import java.net.*;
import java.io.*;
import java.util.regex.*;

public class BankCli {

    public static void main(String[] args) {

		Socket sock = null;              		// client's socket
		InetAddress addr = null;         		// addr of server (local host for now)
		DataOutputStream outStream = null;		// stream used to write to socket
		DataInputStream inStream = null;		//stream used to read server response
		BufferedReader stdin = null;

		System.out.println("Client starting.");

		// set up terminal input
		try {
		    stdin = new BufferedReader(new InputStreamReader(System.in));  
		} catch (Exception e) {
		    System.out.println("Failed to create terminal reader.");
		    System.exit(1);
		}

		// read and send integers over the socket until zero is entered
		boolean exitCmd = false;
		String input = null;
		String response = null;
		do {
			try {
				input = stdin.readLine();
			} catch (Exception e) {
				System.out.println("Failed to read from terminal.");
				System.exit(1);
			}

			// Input verification
			if (input.matches("[CR]<\\d+>|[WD]<\\d+,\\d+>")) {
				//Connect to server
				try {
				    addr = InetAddress.getByName("cormorant.cs.umanitoba.ca");
				    sock = new Socket(addr, 13059);
				    outStream = new DataOutputStream(sock.getOutputStream());
			    	inStream = new DataInputStream(sock.getInputStream());
				} catch (Exception e) {
				    System.out.println("Failed to create socket.");
				    System.exit(1);
				}	

				try {
					outStream.writeUTF(input);
					response = inStream.readUTF();
					System.out.println("[Server]: " + response);
				} catch (Exception e) {
					System.out.println("Communication with the server has failed.");
					System.exit(1);
				}

				//Disconnect from server
				try {
					outStream.close();
					inStream.close();
					sock.close();
				} catch (Exception e) {
					System.out.println("Failed to close socket.");
					System.exit(1);
				}
			} else if (input.matches("E")) {
				exitCmd = true;
			} else {
				System.out.println(input + " is not a recognized command.");
			}
		} while(!exitCmd);

		// close the streams and  socket
		try {
		    stdin.close();
		} catch (Exception e) {
		    System.out.println("Client couldn't close terminal reader.");
		    System.exit(1);
		}

		System.out.println("Client finished.");
    }
}

File: a2/q2/BankCli.py
# Created by:	Colin Dyson
# Student #:	7683407

import sys
import socket

sock = ''
host = "www3.cs.umanitoba.ca"
port = 13059

print("Client starting")

exit = False

while exit == False:
	try:
		term_in = raw_input(">")
	except Exception:
		print("Failed to read from terminal.")
		sys.exit()

	if term_in != "E":
		try:
			sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
			sock.connect((host, port))
		except socket.error as e:
			print("Failed to create socket.", e)
			sys.exit()

		try:
			sock.sendall(term_in + '\n')
			response = sock.recv(1024)
			print("[Server]: " + response)
		except Exception:
			print("Communication with the server has failed.")
			sys.exit()

		try:
			sock.close()
		except Exception:
			print("Failed to close socket")
			sys.exit()

	else:
		exit = True

try:
	sock.close()
except Exception:
	print("Failed to close socket")
	sys.exit()

print("Client finished")
